src/scripts/quests/questTypes/HarvestBerriesQuest.ts

```diff
@@ -16,7 +16,7 @@ class HarvestBerriesQuest extends Quest implements QuestInterface {
 
     public static generateData(): any[] {
         // Getting available Berries (always include Gen 1 Berries)
-        const availableBerries = App.game.farming.berryData.filter(berry => App.game.farming.unlockedBerries[berry.type]() || berry.type < BerryType.Persim);
+        const availableBerries = App.game.farming.berryData.filter(berry => App.game.farming.unlockedBerries[berry.type]() && berry.growthTime[3] < 12000 || berry.type < BerryType.Persim);
```

Comment:
While not strictly needed in this case you may want to add parenthesis around these conditions for the sake of clarity.

```suggestion
        const availableBerries = App.game.farming.berryData.filter(berry => (App.game.farming.unlockedBerries[berry.type]() && berry.growthTime[3] < 12000) || berry.type < BerryType.Persim);
```